src/Workspaces/Core/Portable/Shared/TestHooks/AsynchronousOperationListenerProvider.cs

```diff
@@ -151,7 +151,7 @@ public async Task WaitAllAsync(Workspace? workspace, string[]? featureNames = nu
             do
             {
                 // wait for all current tasks to be done for the time given
-                if (Task.WaitAll(tasks.ToArray(), smallTimeout))
+                if (Task.WaitAll([.. tasks], smallTimeout))
```

Comment:
>[.. tasks] [](http://example.com/codeflow?start=33&length=10)

nit: not sure if this code is high enough frequency to warrant, but with minimal hoop jumps, it looks like this could use your FixedSizeArrayBuilder and cut out one of the array allocations here.